packages/wallet/test/services/DRepRegistrationTracker.test.ts

```diff
@@ -0,0 +1,182 @@
+        certificates: [
+          {
+            ...registerDelegateRepresentativeCertificate,
+            dRepCredential: { hash: foreignDRepKeyHash, type: Cardano.CredentialType.KeyHash }
+          }
+        ]
+      }
+    } as unknown as Cardano.HydratedTx;
+
+    const txWithForeignUnregisterDelegateRepresentativeCertificate = {
+      body: {
+        certificates: [
+          {
+            ...unregisterDelegateRepresentativeCertificate,
+            dRepCredential: { hash: foreignDRepKeyHash, type: Cardano.CredentialType.KeyHash }
+          }
+        ]
+      }
+    } as unknown as Cardano.HydratedTx;
+
+    createTestScheduler().run(({ cold, expectObservable, flush }) => {
+      const target$ = createDRepRegistrationTracker({
+        historyTransactions$: cold('a-b-c-d-e-f-g-h', {
+          a: [],
+          b: [txWithOtherCertificate],
+          c: [txWithOtherCertificate, txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential],
+          d: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate
+          ],
+          e: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate
+          ],
+          f: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate,
+            txWithUnregisterDelegateRepresentativeCertificate_ScriptHashDRepCredential
+          ],
+          g: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate,
+            txWithUnregisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignUnregisterDelegateRepresentativeCertificate
+          ],
+          h: [
+            txWithOtherCertificate,
+            txWithRegisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignRegisterDelegateRepresentativeCertificate,
+            txWithOwnRegisterDelegateRepresentativeCertificate,
+            txWithUnregisterDelegateRepresentativeCertificate_ScriptHashDRepCredential,
+            txWithForeignUnregisterDelegateRepresentativeCertificate,
+            txWithOwnUnregisterDelegateRepresentativeCertificate
+          ]
+        }),
+        pubDRepKeyHash$: of(dRepPublicKeyHash)
+      });
+      expectObservable(target$).toBe('a-b-c-d-e-f-g-h', {
+        a: false,
+        b: false,
```

Comment:
I think we need a `distinctUntilChanged` operator in the DrepRegistrationTracker. We shouldn't emit duplicates imho.

Reply:
done here https://github.com/input-output-hk/cardano-js-sdk/pull/1192/commits/f905e90f5455c5f9df42a4314438796acf479da0#diff-3daae4f884d09049873c2f3c2a5567a39ae0a43af64398a6473983c82c5e0745R50